libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assignment.pass.cpp

```diff
@@ -0,0 +1,253 @@
+    inplace_vector vec{T(0), T(42), T(1337), T(42), T(5)};
+    vec = cuda::std::move(input);
+    assert(!vec.empty());
+    assert(input.size() == 4);
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N> can be move assigned a non-empty input, growing
+    inplace_vector input{T(1), T(42), T(1337), T(0)};
+    inplace_vector vec{T(0), T(42)};
+    vec = cuda::std::move(input);
+    assert(!vec.empty());
+    assert(input.size() == 4);
+    assert(equal_range(vec, expected));
+  }
+}
+
+template <class T>
+__host__ __device__ constexpr void test_init_list()
+{
+  { // inplace_vector<T, 0> can be assigned an empty initializer_list
+    cuda::std::initializer_list<T> input{};
+    cuda::std::inplace_vector<T, 0> vec{};
+    vec = input;
+    assert(vec.empty());
+  }
+
+  using inplace_vector = cuda::std::inplace_vector<T, 42>;
+  cuda::std::initializer_list<T> empty_input{};
+  { // inplace_vector<T, N> can be assigned an empty initializer_list
+    inplace_vector vec{};
+    vec = empty_input;
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N> can be assigned an empty initializer_list, shrinking
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec = empty_input;
+    assert(vec.empty());
+  }
+
+  cuda::std::array<T, 4> expected{T(1), T(42), T(1337), T(0)};
+  { // inplace_vector<T, N> can be assigned a non-empty initializer_list, from empty
+    inplace_vector vec{};
+    vec = {T(1), T(42), T(1337), T(0)};
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N> can be assigned a non-empty initializer_list, shrinking
+    inplace_vector vec{T(0), T(42), T(1337), T(42), T(5)};
+    vec = {T(1), T(42), T(1337), T(0)};
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+
+  { // inplace_vector<T, N> can be assigned a non-empty initializer_list, growing from non empty
+    inplace_vector vec{T(0), T(42)};
+    vec = {T(1), T(42), T(1337), T(0)};
+    assert(!vec.empty());
+    assert(equal_range(vec, expected));
+  }
+}
+
+template <class T>
+__host__ __device__ constexpr void test()
+{
+  test_copy<T>();
+  test_move<T>();
+  test_init_list<T>();
+}
+
+__host__ __device__ constexpr bool test()
+{
+  test<int>();
+  test<Trivial>();
+
+  if (!cuda::std::__libcpp_is_constant_evaluated())
+  {
+    test<NonTrivial>();
+    test<NonTrivialDestructor>();
+    test<ThrowingDefaultConstruct>();
+    test<ThrowingCopyConstructor>();
+    test<ThrowingMoveConstructor>();
+    test<ThrowingCopyAssignment>();
+    test<ThrowingMoveAssignment>();
+  }
+
+  return true;
+}
+
+#ifndef TEST_HAS_NO_EXCEPTIONS
+void test_exceptions()
+{ // assignment throws std::bad_alloc
+  constexpr size_t capacity = 4;
+  using inplace_vector      = cuda::std::inplace_vector<int, capacity>;
+  inplace_vector too_small{};
+
+  try
+  {
+    cuda::std::initializer_list<int> input{0, 1, 2, 3, 4, 5, 6};
+    too_small = input;
```

Comment:
Please assert an exception is actually thrown:
```suggestion
    cuda::std::initializer_list<int> input{0, 1, 2, 3, 4, 5, 6};
    too_small = input;
    assert(false);
```